### PR TITLE
feat(core): convert page state to dom state

### PR DESCRIPTION
**Pull request by blakebyrnes**
This PR changes the name of PageState to DomState and makes it a check for a single state.

- Also removed Hero.on('new'
- DomStateGenerator only compares a single state now
- Bug where Callsite could include internal node files

### CODE REVIEW

File: client/lib/DomState.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+export default class DomState implements IDomState {
+  public name?: string;
+  public url?: string | RegExp;
+  public allTrue: (options: IDomStateAssertions) => void;
+
+  constructor(options: IDomState) {
+    this.name = options.name;
+    this.url = options.url;
+    this.allTrue = options.allTrue;
```
Comment (calebjclark) [inline client/lib/DomState.ts:11]: Does allTrue feel like the right name?

File: client/lib/Hero.ts [hunk 2]
```diff
@@ -481,12 +475,17 @@ export default class Hero extends AwaitedEventTarget<{
...
     options?: Pick<IWaitForOptions, 'timeoutMs'>,
-  ): Promise<keyof T> {
-    return await this.activeTab.waitForPageState(states, options);
+  ): Promise<boolean> {
+    return await this.activeTab.waitForState(state, options);
   }
+
+  public async ensureState(state: IDomState | DomState): Promise<boolean> {
+    return await this.activeTab.ensureState(state);
+  }
```
Comment (calebjclark) [inline client/lib/Hero.ts:487]: We had discussed changing this to `checkState`. I'm okay with either one, just curious your thoughts.

File: client/lib/DomState.ts (hunk 1 above)
Comment (blakebyrnes) [reply client/lib/DomState.ts:11]: It feels a little strange, but I didn't have a better name. We could do something more like "map" syntax- like every, or all

File: client/lib/Hero.ts (hunk 2 above)
Comment (blakebyrnes) [reply client/lib/Hero.ts:487]: I have it changed in the new branch

**Pull request merged**